### PR TITLE
Merge conflicts and PEP build fixed

### PR DESCRIPTION
**Pull request by Purvanshsingh**
reference #456

### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.6.0 and above.

### Current behaviour

Currently #456 having merging conflicts

### New expected behaviour

able to merge.

### CODE REVIEW

**Comment by sameshl** [comment]
Hmm, nice work @Purvanshsingh. Although, I did not quite understand the purpose of this PR atm because currently the #456 is still a POC and not ready to be merged. So, we are not looking to merge it anyway.

**Review by sameshl — COMMENTED** [review]
Hmm, nice work @Purvanshsingh. Although, I did not quite understand the purpose of this PR atm because currently the #456 is still a POC and not ready to be merged. So, we are not looking to merge it anyway.

**Comment by Purvanshsingh** [comment]
@sameshl, There were lots of changes in the hydrus since the PR was opened. So if someone wants to test it locally might face some issue, so I tried to fix them.

File: hydrus/app.py [hunk 1]
```diff
@@ -50,6 +57,43 @@
 # Create a Hydrus app
 app = app_factory(API_NAME)
 socketio = create_socket(app, session)
+
+# global dict to store mapping of each function to be run before
```
Comment (Mec-iS) [inline hydrus/app.py:61]: Docstrings go inside the function declaration using triple quotes `"""`

File: hydrus/app.py [hunk 2]
```diff
@@ -50,6 +57,43 @@
...
+        global before_request_funcs
+        before_request_funcs[path][method] = f
+        return f
+
+    return wrapper
+
+
+@custom_before_request('/api/MessageCollection', 'PUT')
+def do_this_before_put_on_drone_collections():
+    print("Do something before PUT request on MessageCollection")
```
Comment (Mec-iS) [inline hydrus/app.py:89]: we cannot have `print` statements merged in the main branch

File: hydrus/app.py [hunk 3]
```diff
@@ -58,17 +102,17 @@
...
-            # Start the Hydrus app
-            http_server = WSGIServer(('', PORT), app)
-            logger.info(f'Running server at port {PORT}')
-            try:
-                http_server.serve_forever()
-            except KeyboardInterrupt:
-                pass
+        set_api_name(app, API_NAME), set_doc(app, apidoc), \
+        set_hydrus_server_url(app, HYDRUS_SERVER_URL), set_session(app, session):
+    if __name__ == "__main__":
```
Comment (Mec-iS) [inline hydrus/app.py:107]: these changes will make the conditionals to run outside of the `with` statement

**Comment by Mec-iS** [comment]
These changes are not useful

**Pull request closed without merging**